sacrebleu/compat.py

```diff
@@ -40,13 +38,14 @@ def corpus_bleu(sys_stream: Union[str, Iterable[str]],
 
 def raw_corpus_bleu(sys_stream,
                     ref_streams,
-                    smooth_value=None) -> BLEUScore:
+                    smooth_value=0) -> BLEUScore:
     """Convenience function that wraps corpus_bleu().
```

Comment:
I think we should factor out the 0 here into a global constant, e.g., `SMOOTH_VALUE_DEFAULT`.

Reply:
The defaults for BLEU are already defined in `BLEU.SMOOTH_DEFAULTS`. The reason it is hard-coded to be 0 here was to preserve backward compatibility with the `raw_corpus_bleu` of older sacreBLEU versions. But since we're breaking API here, I can revert this to be `None` so that it defaults to the new internal default of 0.1 ?

Reply:
The alternative is to remove this "convenience" function completely and expect people to use corpus_bleu with arguments that they'd like to use.

Reply:
So `raw_corpus_bleu` hard-codes a `smooth_value` of `floor`. Since the API allows the value to be changed, I would suggest that instead of `smooth_value=0`, we use `smooth_value=BLEU.SMOOTH_DEFAULTS["floor"]`. That way the value is explicit, instead of potentially getting lost here. What do you think of this?

Reply:
I'd like to keep the convenience function. I think Sockeye might be using it.

Reply:
`smooth_value=BLEU.SMOOTH_DEFAULTS["floor"]` is equivalent (although more explicit) to the unmodified version of the signature which was `smooth_value=None` (a `None` fetches the default from that very same dictionary internally in `Bleu` class)

The question here is whether we want sacreBLEU >= 1.5 to give the same sentence bleu scores as sacreBLEU < 1.5 when using this function, or not. I think we don't care as of now since we already introduced other behavior differences as well.

So I'll make this `BLEU.SMOOTH_DEFAULTS["floor"]` to be more readable.

The reason why I suggested this function is that, it again falls back to `floor` although all other methods for sentence-bleu are now synchronized to use `exp` smoothing. It's kind of weird the choices here =)